homeassistant/components/fan/dyson.py

```diff
@@ -0,0 +1,176 @@
+"""Support for Dyson Pure Cool link fan."""
+import logging
+from homeassistant.components.fan import (FanEntity, SUPPORT_OSCILLATE,
+                                          SUPPORT_SET_SPEED,
+                                          SUPPORT_NIGHT_MODE,
+                                          SUPPORT_AUTO_MODE)
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.components.dyson import DYSON_DEVICES
+
+DEPENDENCIES = ['dyson']
+REQUIREMENTS = ['libpurecoollink==0.1.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Dyson fan components."""
+    _LOGGER.info("Creating new Dyson fans")
+    devices = []
+    # Get Dyson Devices from parent component
+    for device in hass.data[DYSON_DEVICES]:
+        devices.append(DysonPureCoolLinkDevice(hass, device))
+    add_devices(devices)
+
+
+class DysonPureCoolLinkDevice(FanEntity):
+    """Representation of a Dyson fan."""
+
+    def on_message(self, message):
+        """Called when new messages received from the fan."""
+        _LOGGER.debug("Message received for fan device %s : %s", self.name,
+                      message)
+        if self.hass and self.entity_id:
```

Comment:
`if self.hass` should not needed or you have done a desine misstake. Make sure that your device is receive callback after `HOMEASSISTANT_START` or register your device callback inside `async_added_to_hass`.